src/scripts/records/NotificationRecord.js

```diff
@@ -0,0 +1,25 @@
+/* @flow */
+
+import { Record as ImmutableRecord } from 'immutable'
+
+class Notification extends ImmutableRecord({
+  title: '',
+  message: '',
+  position: 'tr',
+  autoDismiss: 0,
+  action: {
+    label: '',
+    callback: null
+  }
+}) {
+  title: string
+  message: string
+  position: string
+  autoDismiss: number
+  action: {
+    label: '',
+    callback: () => void
```

Comment:
What is `callback`? I think it would be better to rename this to indicate when it is called. 

What about `onPerformAction`?

Reply:
Is a generic callback, this is just a default!

Reply:
What is a generic callback? That is the problem, it is very unclear what this is supposed to do